### ISSUE TITLE
Pets do not keep up with their masters redux

### ISSUE THREAD

**Issue by Stence** [elysium-project/server#579]
This is basically a copypaste of my bugtracker ticket that I created which you can find here: https://elysium-project.org/bugtracker/issue/1297
It has been confirmed by the QA and has over 60 votes. 

"Pets do not keep up with their masters like they did in vanilla WoW. The pet always seems to be lagging behind instead of being in sync with the master's movements. Pets also slow down when entering combat.

Example of all of this: https://www.youtube.com/watch?v=UPjqlhyq4Vc

The pet should keep up with the master. Several classic WoW videos below illustrate this mechanic quite well.

Example 1 : https://www.youtube.com/watch?v=7NagvKFLngU Pay attention to the way the fel hunter moves when the master moves. Key points: 1:12 , 7:08

Example 2 : https://www.youtube.com/watch?v=jFf_FOQ48qY Pay attention once again to the way the warlock's pet moves. Even at the beginning the pet is able to keep up with the mounted warlock. Key points: 0:05, 3:05, 4:31

Example 3 : https://youtu.be/EQqLRpwrUr8?t=32 A warlock's enslaved demon pet keeps up with him entirely."

The biggest problem I experience with pet responsiveness is if I dismount to attack something the pet is still trying to catch up to me. Very annoying if I am trying to attack a player and I need my felhunter on the frontlines immediately in order to silence a target.

Also another problem is that whenever I am traveling through a mob dense zone like silithus my pet is constantly being slowed down by entering combat from nearby mobs. Either I wait for the pet to catch up or run far ahead enough on my mount that the pet is abandoned and needs to be resummoned.

**Comment by thepparker** [comment]
I've investigated this a little. Unfortunately, increasing the responsiveness of pets comes at a performance cost and with thousands of players all using pets, a rather considerable one. This is why you see in those later Netherwing videos the pets following a little further behind. That being said, I believe there is a middle ground where pets can be made to follow slightly closer but to have it at retail levels is unfeasible. They will always lag a little behind,

**Comment by sknnbones** [comment]
@thepparker Will this be the same with your updated "wild stats" pet commit? [movespeed mainly]

**Comment by thepparker** [comment]
No, the move speed is handled a bit differently when in follow mode compared to chase mode. They obviously aren't running at normalized speed while you are mounted, for example. 

If merged, you can rest assured that hunter pets will hunt people down.

**Comment by GuybrushGit** [comment]
@thepparker could this help this issue?
https://github.com/TrinityCore/TrinityCore/pull/8753/files

**Comment by thepparker** [comment]
Speedup when at a long (> 7 yards) distance is already implemented. The problem is we don't make the pet move until it's a certain distance from the owner, and we also don't update the path frequently enough to ensure that the pet stays closer. Likewise, if we were to make them run faster at a closer range they reach the path end point before it is updated again so they appear to stutter.
